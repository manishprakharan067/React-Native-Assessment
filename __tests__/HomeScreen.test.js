import React from 'react';
import {Platform, NativeModules} from 'react-native';
import HomeScreen from '../screens/Home';
import Data from '../Data.json';
import configureStore from 'redux-mock-store';
import {render} from '@testing-library/react-native';

const INITIAL_STATE = {ResultData: Data.results};
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

jest.useFakeTimers();
test('renders correctly', async () => {
  Platform.OS = 'android';
  NativeModules.InternetCon = {
    getInternetStatus: jest.fn(),
  };
  render(<HomeScreen {...INITIAL_STATE} />);
});
