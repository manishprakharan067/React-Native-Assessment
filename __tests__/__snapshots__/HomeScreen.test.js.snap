// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginTop": 0,
    }
  }
>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
  <ActivityIndicator
    color="red"
    size="large"
    style={
      Object {
        "left": 357.1428571428571,
        "position": "absolute",
        "top": 667,
      }
    }
  />
</RCTSafeAreaView>
`;
